### ISSUE TITLE
GitLab repository listing

### ISSUE THREAD

**Issue by gal-legit** [Legit-Labs/legitify#162]
### TL;DR

Need to fix GitLab support for different use cases

### Expected behavior

Always list everything that is possible

### Observed behavior

Some use cases aren't covered

### Version

v0.0.23

### On which operating system are you using legitify?

Linux

### Additional information

Need to cover the following matrix:
- PAT of a site admin
- the --org flag is provided
- the --repo flag is provided

Need to make sure to list:
- All repos for groups where the PAT is either an owner or a site admin
- All repos for users where the PAT is either of the user or of a site admin
- All repos where the PAT is not owner/admin of group but has sufficient permissions for the repo
Also fix the list-repos command to be aligned to the fixed behavior of the analyze command
